### PR TITLE
fix: Properly restore kv.spec.uninstallStrategy

### LINKED ISSUE #12730: check-tests-for-flakes lane was constantly failing on PR changing Operator tests
On the PR https://github.com/kubevirt/kubevirt/pull/12616 the check-tests-for-flakes lane was constantly failing [1]. This might indicate a flakiness present in the operator tests.

Background: the check-tests-for-flakes lane runs tests in a random order, repeated by five times . This has shown to reveal 85% of flaky tests [2].

_Originally posted by @dhiller in https://github.com/kubevirt/kubevirt/issues/12616#issuecomment-2328293137_

[1]: https://prow.ci.kubevirt.io/pr-history/?org=kubevirt&repo=kubevirt&pr=12616
[2]: https://github.com/kubevirt/kubevirt/blob/main/automation/repeated_test.sh#L21

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
Currently serial tests rely on the global cleanup. Actually the global cleanup just ensure the kv.spec.config is restored, not the entire spec.
Test id [test_id:3683] changes the uninstallStrategy to `BlockUninstallIfWorkloadsExist` but does not restore the previous setting.
This can lead to failure to the subsequent tests, for example `[test_id:3146]should be able to delete and re-create kubevirt install`, in which uninstall strategy is evaluated.

Failure example: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/13139/pull-kubevirt-check-tests-for-flakes/1854890798374457344

/kind flake

Fixes #12730

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by akalenyu — COMMENTED** [review]
Nice catch! should we maybe ensure kv is completely restored in global cleanup?
I am afraid this is a similar symptom to what I bugged you about in https://github.com/kubevirt/kubevirt/pull/12766

**Comment by fossedihelm** [comment]
> Nice catch! should we maybe ensure kv is completely restored in global cleanup? I am afraid this is a similar symptom to what I bugged you about in #12766

I was tempted to do so, but I have preferred to do so in a separate wider PR and focus on the simple bug fix.

**Comment by akalenyu** [comment]
> > Nice catch! should we maybe ensure kv is completely restored in global cleanup? I am afraid this is a similar symptom to what I bugged you about in #12766
> 
> I was tempted to do so, but I have preferred to do so in a separate wider PR and focus on the simple bug fix.

Okay sure. Just my couple of cents - using patch here, I believe we'll be missing hotloop/test bugs like the one you're describing. But that's a whole different discussion and I am willing to accept that the operator changes the CR at it's discretion.
/lgtm

File: tests/operator/operator.go [hunk 1]
```diff
@@ -1411,19 +1411,34 @@ spec:
...
+				}, 60*time.Second, time.Second).Should(Equal(v1.KubeVirtUninstallStrategyBlockUninstallIfWorkloadsExist))
+			})
 
+			AfterEach(func() {
+				By("cleaning the uninstall strategy")
+				patchBytes, err := patch.New(patch.WithRemove("/spec/uninstallStrategy")).GeneratePayload()
+				Expect(err).ToNot(HaveOccurred())
+				_, err = virtClient.KubeVirt(originalKv.Namespace).Patch(context.Background(), originalKv.Name, types.JSONPatchType, patchBytes, metav1.PatchOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() (v1.KubeVirtUninstallStrategy, error) {
```
Comment (xpivarc) [inline tests/operator/operator.go:1435]: This should not be needed?

**Review by xpivarc — COMMENTED** [review]
/approve
We need to solve this for whole suite but I don't mind doing this patch first

File: tests/operator/operator.go (hunk 1 above)
Comment (akalenyu) [reply tests/operator/operator.go:1435]: > Currently serial tests rely on the global cleanup. Actually the global cleanup just ensure the kv.spec.config is restored, not the entire spec.

**Pull request merged**